### DISCUSSION TITLE
What is SemanticURI?

### DISCUSSION THREAD

**Discussion by GaryKopp** [eclipse-emfcloud/emfcloud#88] [All other] [answered]
ecore-glsp code makes references to something called a SemanticURI. Is there a definition of that that is? I'm guessing, possibly wildly, that it's a concept coming from GLSP.

**Comment by planger** [comment, marked as answer]
The semantic URI identifies the model element. So it is an identifier that points to the model element that the respective graphical element represents in the diagram.

In diagrams you typically have graphical elements (nodes, edges, etc.) which represent model elements from the domain model (e.g. EAttributes, EClasses in the context of Ecore). The domain model is often referred to as *semantic model* and the model with the graphical elements *notation model*. I believe this terminology comes from [GMF](https://wiki.eclipse.org/Graphical_Modeling_Framework/FAQ#What_is_the_relationship_between_semantic.2C_domain.2C_notation.2C_and_graphical_definition_models.3F). Partly this terminology is also applied in some diagram editor implementations done with GLSP, e.g. with semantic URI as the URI that points to the semantic model element
